services/core/shared/core/compat/sdk_v5/blocks.js

```diff
@@ -129,34 +230,33 @@ const buildIndex = async (from, to) => {
 	const numOfPages = Math.ceil((to + 1) / MAX_BLOCKS_LIMIT_PP - from / MAX_BLOCKS_LIMIT_PP);
 
 	const highestIndexedHeight = await blocksCache.get('highestIndexedHeight');
-	Array(numOfPages).fill().forEach(async (_, pageNum) => {
+	for (let pageNum = 0; pageNum < numOfPages; pageNum++) {
 		const pseudoOffset = to - (MAX_BLOCKS_LIMIT_PP * (pageNum + 1));
 		const offset = pseudoOffset > from ? pseudoOffset : from - 1;
 		logger.info(`Attempting to cache blocks ${offset + 1}-${offset + MAX_BLOCKS_LIMIT_PP}`);
-		// TODO: Revert to standard notation, once getBlocks is fully implemented
-		// const blocks = await getBlocks({
-		// 	limit: MAX_BLOCKS_LIMIT_PP,
-		// 	offset: offset - 1,
-		// 	sort: 'height:asc',
-		// });
-		// TODO: Add check when below call fails similar to SDKv4
-		const blocks = await getBlocks({
-			heightRange: { from: offset + 1, to: offset + MAX_BLOCKS_LIMIT_PP },
-		});
-		await indexBlocks(blocks.data);
 
-		blocks.data = blocks.data.sort((a, b) => a.height - b.height);
+		/* eslint-disable no-await-in-loop */
+		let blocks;
+		do {
+			blocks = await getBlocksByHeightBetween(offset + 1, offset + MAX_BLOCKS_LIMIT_PP);
+		} while (!(blocks.length && blocks.every(block => !!block && !!block.height)));
+
+		await indexBlocksQueue.add('indexBlocksQueue', { blocks });
+
+		const sortedBlocks = blocks.sort((a, b) => a.height - b.height);
 
-		const topHeightFromBatch = (blocks.data.pop()).height;
-		const bottomHeightFromBatch = (blocks.data.shift()).height;
+		const topHeightFromBatch = (sortedBlocks.pop()).height;
+		const bottomHeightFromBatch = (sortedBlocks.shift()).height;
 		const lowestIndexedHeight = await blocksCache.get('lowestIndexedHeight');
-		if (bottomHeightFromBatch < lowestIndexedHeight) await blocksCache.set('lowestIndexedHeight', bottomHeightFromBatch);
+		if (bottomHeightFromBatch < lowestIndexedHeight || lowestIndexedHeight === 0) await blocksCache.set('lowestIndexedHeight', bottomHeightFromBatch);
```

Comment:
This part can be done using MySQL index.

The algorithm should find missing blocks in the index during start and retrieve those missing blocks with their transactions, accounts, etc. The top height of the finalized blocks could be stored in the db to avoid re-scanning all blocks during start.

Reply:
I believe we should create a separate issue for this change. The amount of changes and the time required for this is a lot to include in the same PR.

Reply:
#360